app/models/tasklist/ReadsWrites.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.tasklist
+
+import play.api.libs.json.{Json, OWrites, Reads}
+
+trait ReadsWrites[A] {
+  // Allows to read and write a case class without parameters.
+  implicit val nonStrictReads: Reads[A] = Reads.pure(asInstanceOf[A])
```

Comment:
Using asInstanceOf[A] can result in runtime exceptions if the types do not match. I would suggest to user enumeratum library. That will even simplify this code further and we will not have to deal with these possible errors